### PR TITLE
Add Github Actions for Format + Testing

### PR DESCRIPTION
**Pull request by srvasude**
Github Action for CI.

- I've removed the formatting action, because there were version diffs on what github was installing and what's at Google and was getting different results when I tried to keep these as close as possible (Google's depends on Clang 19.0 for example which isn't released yet).

- I've added a .clang-format file to at least make these changes uniform.

- Fixed tests at HEAD (adapter_test was broken by recent normal changes).

- Clang-formatted rest of code base.

### CODE REVIEW

File: cxx/distributions/adapter_test.cc [hunk 1]
```diff
@@ -20,8 +20,8 @@ BOOST_AUTO_TEST_CASE(adapt_normal) {
   ad.incorporate("7.0");
   ad.unincorporate("-2.0");
 
-  BOOST_TEST(ad.logp("6.0") == -3.1331256657870137, tt::tolerance(1e-6));
-  BOOST_TEST(ad.logp_score() == -4.7494000141508543, tt::tolerance(1e-6));
+  BOOST_TEST(ad.logp("6.0") == -2.7673076255063034, tt::tolerance(1e-6));
```
Comment (emilyfertig) [inline cxx/distributions/adapter_test.cc:23]: Do you know why this changed?

File: cxx/tests/test_util_math.cc
Comment (emilyfertig) [inline cxx/tests/test_util_math.cc:1]: Was this moved somewhere else?

**Review by emilyfertig — APPROVED** [review]

File: cxx/distributions/adapter_test.cc (hunk 1 above)
Comment (srvasude) [reply cxx/distributions/adapter_test.cc]: The tests weren't passing after changes to normal (to have double mean/stddev) so I updated the tests here.

File: cxx/tests/test_util_math.cc
Comment (srvasude) [reply cxx/tests/test_util_math.cc:1]: Deleted. We had a util_math_test which subsumes this.

**Pull request merged**